ee/clickhouse/queries/groups_join_query.py

```diff
@@ -1,9 +1,19 @@
 from typing import Dict, List, Optional, Set, Tuple, Union
 
+from ee.clickhouse.materialized_columns.columns import ColumnName
 from ee.clickhouse.queries.column_optimizer import ColumnOptimizer
 from posthog.models import Filter
 from posthog.models.filters.path_filter import PathFilter
 from posthog.models.filters.retention_filter import RetentionFilter
+from posthog.models.property import GroupTypeIndex
+
+CREATED_AT = ColumnName("created_at")
+GROUP_PROPERTIES = ColumnName("group_properties")
+ACCEPTED_EXTRA_FIELDS = [CREATED_AT]
+GROUP_ALIASES = {
```

Comment:
I added this type of query building so that [this select](https://github.com/PostHog/posthog/pull/7144/files#diff-68ee5fd398e73fa594adb5050743b2d33a3e9e407ca1444370432c49a18e6677R52) in the above query will be flexible and able to account for different fields of different group types. 

Also, if there's a query that's filtering on multiple group types, there will be multiple group joins so `Dict[GroupTypeIndex, ColumnName]` type allows you to specify which group join to add the fields in and for the query above to select the right one. 

Open to simplifying as you suggest but I think it would make the select in the outer query have more assumptions with regards to which group type index to select on

Reply:
> I think it would make the select in the outer query have more assumptions with regards to which group type index to select on

Let's take a look at the usecase:

```python
            join_queries.append(
                f"""
                INNER JOIN (
                    SELECT
                        group_key,
                        argMax(group_properties, _timestamp) AS group_properties_{group_type_index}
                        {(
                            " ".join(
                                f", argMax({field}, _timestamp) AS {GROUP_ALIASES[field]}{group_type_index}"
                                for field in columns_to_add
                            )
                        )}
                    FROM groups
                    WHERE team_id = %(team_id)s AND group_type_index = %({var})s
                    GROUP BY group_key
...
```

The problem I see it's this constant makes it becomes almost impossible to understand what this will expand to. One would need to:
- Jump to see what `_extra_fields` are passed
- Jump to see what `ACCEPTED_EXTRA_FIELDS` are, do the set logic
- Jump to constant to see what the mapping is
- Jump to query and see we're implicitly appending {group_type_index} at the end of it

We'd be jumping between so many different files for confusing reasons, making this hard to grok.

Let's discuss a potential solution. Perhaps we should be doing 

```
SELECT columns
FROM groups
WHERE id IN ({events_query})
```

instead and not bring this complexity within event queries and into all classes in the first place?

Reply:
Resolved on call: We decided to always just retrieve IDs of groups or persons and retrieve them using postgres data. This way it's consistent across all insight implementations and doesn't require as much query reshaping to retrieve the right fields